Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/UserManager.java

```diff
@@ -335,6 +336,58 @@ public boolean addNewUser(AccountRequest request) throws QuadrigaStorageExceptio
         
         return success;
     }
+    
+    
+    /**
+     * This method adds a new request to access quadriga for the user authenticated by a social provider e.g. github, facebook etc.
+     * 
+     * @param username
+     *            The user name of the user who needs access to quadriga
+     * @param fullname
+     *            The full name of the user who needs access to quadriga     
+     * @param email
+     *            The email id of the user who needs access to quadriga
+     * @param provider
+     *            The name of the social provider that authenticated the user for quadriga
+     * @param userIdOfProvider
+     *            The user id of the user in the domain of the social provider that authenticated the user for quadriga
+     * @return true if the user approval request was placed successfully , 
+     *         false if the user approval request could not be placed or the user request already exists (avoid duplicate requests)
+     * 
+     * @author Chiraag Subramanian
+     * @throws QuadrigaStorageException, UsernameExistsException
+     */
+    @Override
+    @Transactional
+    public boolean addSocialUser(String username, String fullname, String email, String provider, String userIdOfProvider) throws QuadrigaStorageException, UsernameExistsException {
```

Comment:
I don't think you need this method. Just reuse the addNewUser method and make required changes.